S/spdlog/build_tarballs.jl

```diff
@@ -0,0 +1,31 @@
+using BinaryBuilder
+
+name = "spdlog"
+version = v"1.14.1"
+
+sources = [GitSource("https://github.com/gabime/spdlog.git", "27cb4c76708608465c413f6d0e6b8d99a4d84302")]
+
+script = raw"""
+cd ${WORKSPACE}/srcdir/spdlog*
+mkdir build
+cd build
+cmake -S .. -B . \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_PREFIX_PATH=${prefix} \
+    -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DSPDLOG_FMT_EXTERNAL=ON \
+    -DSPDLOG_BUILD_SHARED=ON \
+    -DSPDLOG_BUILD_PIC=ON \
+    -DSPDLOG_BUILD_EXAMPLE=OFF
+make -j${nproc} install
+"""
+
+platforms = expand_cxxstring_abis(supported_platforms())
+
+products = [LibraryProduct("libspdlog", :libspdlog)]
+
+dependencies = [Dependency("Fmt_jll"; compat="11")]
```

Comment:
Evidently the `compat` specification here doesn't work how I thought

Reply:
https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.Dependency

> The optional keyword argument `compat` can be used to specify a string for use in the `Project.toml` of the generated Julia package. If `compat` is non-empty and `build_version` is not passed, the latter defaults to the minimum version compatible with the `compat` specifier.

First existing version of `Fmt_jll` in the v11 seres is 11.0.1:
```suggestion
dependencies = [Dependency("Fmt_jll"; compat="11.0.1")]
```

Reply:
IIUC, that will build spdlog with 11.0.1, but it will allow spdlog_jll to be used with Fmt_jll 11.0.2. Is that right?

Reply:
Yes, that's standard Pkg compat specification.

Reply:
Right, I guess what I was confirming that it uses the minimum version within the declared range of compatibility for building (almost like inverse Pkg compat behavior), but maximum within the declared range otherwise (standard Pkg compat).